### PR TITLE
Migrate to dotfiles to Nix

### PR DESCRIPTION
**Pull request by bphenriques**
# Description

Migrates to `nix` managed environment leveraging `nix`, `nix flake`, and `homebrew` (MacOS) in order to have a _more_ reproducible environment.

- [x] Pin zinit and its packages - If not possible, the likeness of breaking is low but defeats the purpose. See #2. 
- [ ] Set `zsh` from `nixpkgs` as the default shell. See #3 .
- [ ] Does it make sense to include CI? See #4 
- [X] Review unused stuff.

# Questions

- After updating `flakes.lock`, how to can I know the set of `pkgs` that I use that were updated and if they have breaking changes?

# Test Plan

Status: ✅ 

_Oh god_: Format MacOS and perform the instructions described in the `README.md`. Check that all the tools are working correctly, including GPG signing which can be fuzzy to set sometimes. Also setup on my work laptop.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-# Manually stow files: make module=<module> stow
-stow:
-	stow --dir $(MAKEFILE_LOCATION) --target $(HOME) $(module)
+build: doctor
+	home-manager build
 
-# Check for bogus links
-doctor:
-	chkstow --target $(HOME) --badlinks 2>/dev/null | grep -v ".*/Library/.*"
+sync: doctor
```
Comment (bphenriques) [inline Makefile:9]: With flakes, one needs to run `./result/activate`

**Comment by malob** [comment]
I saw this PR via the mention on my `nix-darwin` PR for the Homebrew module.

Just wanted to drop a quick note to say that my Nix config repo is a flake that provides a `darwinModules` output, which includes a version of the `homebrew` module that I keep up to date with the version on the PR: https://github.com/malob/nixpkgs/blob/master/flake.nix

**Comment by bphenriques** [comment]
Thank you @malob! Will take a look

**Comment by bphenriques** [comment]
> I saw this PR via the mention on my `nix-darwin` PR for the Homebrew module.
> 
> Just wanted to drop a quick note to say that my Nix config repo is a flake that provides a `darwinModules` output, which includes a version of the `homebrew` module that I keep up to date with the version on the PR: https://github.com/malob/nixpkgs/blob/master/flake.nix

👋 I only grasped later what you meant! I am now using your `darwinModules` output as opposed to pointing my `nix-darwin` input to your profile 🙄 Thank you! 👍

**Pull request merged**